src/shared/clean-ipc.c

```diff
@@ -41,8 +41,20 @@
 #include "macro.h"
 #include "string-util.h"
 #include "strv.h"
+#include "user-util.h"
 
-static int clean_sysvipc_shm(uid_t delete_uid) {
+static bool match_uid_gid(uid_t subject_uid, gid_t subject_gid, uid_t delete_uid, gid_t delete_gid) {
+
+        if (uid_is_valid(delete_uid) && subject_uid == delete_uid)
+                return true;
```

Comment:
Well, the kernel actually isn't very good at prohibiting the UIDs/GIDs (uint16_t) -1 and (uint32_t) -1. Hence, in some weird case /proc/sysvipc/shm and friends might actually list them as owners, and I justed wanted to make sure that passing UID_INVALID really always means "disable matching by UID" instead of "match against UID -1". Hence the explicit check.